Add the 'Getting started with DateBox' article (#1004)
DevExpress/devextreme-documentation: concepts/05 Widgets/DateBox/00 Getting Started with DateBox/00 Getting Started with DateBox.md

#include tutorials-intro-installationnote

**DateBox** is a widget that allows users to set a certain date, time, or their combination.

This tutorial shows how to configure basic **DateBox** features. The created widget allows setting the date and time from the allowed date range, logs the set value to console, and prevents users from specifying the US bank holidays and weekend days.

<div class="simulator-desktop-container" data-view="/Content/Applications/20_1/GettingStartedWith/DateBox/index.html, /Content/Applications/20_1/GettingStartedWith/DateBox/index.js, /Content/Applications/20_1/GettingStartedWith/DateBox/index.css"></div>

> Refer to the subtopics for details on every configuration step. You can also see the full code below or download it from [this GitHub repository](https://github.com/DevExpress-Examples/getting-started-with-datebox):

Reviewer: The link should open in a new tab.

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#dateBoxContainer").dxDateBox({
            type: "datetime",
            max: now,
            min: new Date(1900, 0, 1),
            value: now,
            onValueChanged: function(data) {
                console.log(data.value);
                console.log(data.previousValue);
            },
            disabledDates: function(args) {
                const dayOfWeek = args.date.getDay();
                const isWeekend = args.view === "month" && (dayOfWeek === 0 || dayOfWeek === 6); 
                return isWeekend || isHoliday(args.date);
            }
        });

        function isHoliday(date) {

>             for (let holiday of holidays) {

Reviewer: The array of holidays is missing in all technologies.

                if (date.getDate() === holiday.getDate() && date.getMonth() === holiday.getMonth()) {
                    return true;
                }          
            }
            return false;
        }
    });

    <!-- tab: index.html -->
    <!DOCTYPE html>
    <html>
        <head>
            <script type="text/javascript" src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
            <link rel="stylesheet" href="https://cdn3.devexpress.com/jslib/20.1.4/css/dx.common.css">
            <link rel="stylesheet" href="https://cdn3.devexpress.com/jslib/20.1.4/css/dx.light.css">
            <script type="text/javascript" src="https://cdn3.devexpress.com/jslib/20.1.4/js/dx.all.js"></script>
            <script type="text/javascript" src="index.js"></script>
        </head>
        <body class="dx-viewport">
            <div id="dateBox"></div>
        </body>
    </html>

##### Angular

    <!-- tab: app.component.html -->
    <dx-date-box    
        type="datetime"
        [min]="minDate"
        [max]="now"
        [value]="now"
        (onValueChanged)="onValueChanged($event)"

>         [disabledDates]="getDisabledDates"
>     >

Reviewer: ```suggestion
        [disabledDates]="getDisabledDates">
```

    </dx-date-box>

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core';

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        title = 'Getting started with DateBox';
        holidays: Date[];
        now: Date = new Date();
        minDate: Date = new Date(1900, 0, 1);

>         constructor(service: AppService){

Reviewer: ```suggestion
        constructor(service: AppService) {
```

            this.holidays = service.getHolidays();
            this.getDisabledDates = this.getDisabledDates.bind(this);
        }

>         onValueChanged(e){

Reviewer: ```suggestion
        onValueChanged(e) {
```

            console.log(e.previousValue);
            console.log(e.value);
        }

        getDisabledDates(args): boolean {
            const dayOfWeek = args.date.getDay();
            const isWeekend = args.view === "month" && (dayOfWeek === 0 || dayOfWeek === 6 );
            return isWeekend || this.isHoliday(args.date);
        }

        isHoliday(date): boolean {
            for (let holiday of this.holidays) {
                if (date.getDate() === holiday.getDate() && date.getMonth() === holiday.getMonth()) {
                    return true;
                }          
            }
            return false;
        }
    }

    <!-- tab: app.module.ts -->
    import { BrowserModule } from '@angular/platform-browser';
    import { NgModule } from '@angular/core';
    import { AppComponent } from './app.component';

    import { DxDateBoxModule } from 'devextreme-angular';

    @NgModule({
        declarations: [
            AppComponent
        ],
        imports: [
            BrowserModule,
            DxDateBoxModule
        ],

>         providers: [ ],

Reviewer: I think the data service providing holidays should be imported and registered in the module.

Author: The service imported in app.component.ts and not in the app.module.ts should suffice. See:
https://js.devexpress.com/Documentation/Guide/Widgets/ButtonGroup/Getting_Started_with_ButtonGroup/
https://js.devexpress.com/Demos/WidgetsGallery/Demo/SelectBox/SearchAndEditing/Angular/Light/

            :min="minDate"
            :max="now"
            :value="now"
            @value-changed="onValueChanged"
            :disabled-dates="getDisabledDates"
        />
    </template>

    <script>
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { DxDateBox } from 'devextreme-vue/date-box';

    export default {
        components: {
            DxDateBox
        },
        data() {
            return {
                now: Date(),
                minDate: new Date(1900, 0, 1)
            }
        },
        methods: {
            onValueChanged(e) {
                console.log(e.previousValue);
                console.log(e.value);
            },
            getDisabledDates(args) {
                const dayOfWeek = args.date.getDay();
                const isWeekend = args.view === "month" && (dayOfWeek === 0 || dayOfWeek === 6 );
                return isWeekend || this.isHoliday(args.date);
            },
            isHoliday(date) {
                for (let holiday of holidays) {
                    if (date.getDate() === holiday.getDate() && date.getMonth() === holiday.getMonth()) {

>                     return true;

Reviewer: ```suggestion
                        return true;
```

#include tutorials-intro-installationnote

> **DateBox** is a widget that allows users to set a certain date, time, or their combination.

Reviewer 2: ```suggestion
**DateBox** is a widget that allows users to set a certain date, time, or date/time combination.
```

> This tutorial shows how to configure basic **DateBox** features. The created widget allows setting the date and time from the allowed date range, logs the set value to console, and prevents users from specifying the US bank holidays and weekend days.

Reviewer 2: ```suggestion
This tutorial shows how to configure basic **DateBox** features. The newly created widget allows users to set the date and time from a specific date range, logs this value to the console, and prevents users from specifying weekends and US bank holidays.
```

> This tutorial shows how to configure basic **DateBox** features. The newly created widget allows users to set the date and time from a specific date range, logs this value to the console, and prevents users from specifying weekends and US bank holidays.

Reviewer 2: ```suggestion
This tutorial shows how to configure basic **DateBox** features. The newly created widget allows users to set the date and time from a specific date range, logs this value to the console, and prevents users from specifying weekend days (Saturday and Sunday) and US bank holidays.
```

<div class="simulator-desktop-container" data-view="/Content/Applications/20_1/GettingStartedWith/DateBox/index.html, /Content/Applications/20_1/GettingStartedWith/DateBox/index.js, /Content/Applications/20_1/GettingStartedWith/DateBox/index.css"></div>

> Refer to the subtopics for details on every configuration step. You can also see the full code below or download it from <a href="https://github.com/DevExpress-Examples/getting-started-with-datebox" target="_blank">this GitHub repository</a>:

Reviewer 2: ```suggestion
Refer to the subtopics for details on each configuration step. You can also see the full code below or download it from the following GitHub repository: <a href="https://github.com/DevExpress-Examples/getting-started-with-datebox" target="_blank">getting-started-with-datebox</a>.
```

Reviewer 2: "Refer to the subtopics for details on each configuration step."

Where are the links to these subtopics located?

Author: The links are in the left panel:
![image](https://user-images.githubusercontent.com/66619750/87015864-a1352c00-c1d6-11ea-89db-47b4054b810a.png)

Reviewer 2: Thank you for the clarification.

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        const now = new Date();
        const holidays = [
            new Date(0, 0, 1),
            new Date(0, 0, 20),
            new Date(0, 1, 17),
            new Date(0, 4, 10),
            new Date(0, 4, 25),
            new Date(0, 5, 21),
            new Date(0, 6, 4),
            new Date(0, 8, 7),
            new Date(0, 9, 5),
            new Date(0, 9, 12),
            new Date(0, 10, 11),
            new Date(0, 10, 26),
            new Date(0, 10, 27),
            new Date(0, 11, 24),
            new Date(0, 11, 25),
            new Date(0, 11, 31)
        ];

        $("#dateBoxContainer").dxDateBox({
            type: "datetime",
            max: now,
            min: new Date(1900, 0, 1),
            value: now,
            onValueChanged: function(data) {
                console.log(data.value);
                console.log(data.previousValue);
            },
            disabledDates: function(args) {
                const dayOfWeek = args.date.getDay();
                const isWeekend = args.view === "month" && (dayOfWeek === 0 || dayOfWeek === 6); 

>                 return isWeekend || isHoliday(args.date);

Reviewer 3: Don't forget the previous note.
